src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -167,4 +167,61 @@ protected function instantiateController($class)
     {
         return new $class();
     }
+
+    private function getControllerError($callable)
+    {
+        if (is_string($callable)) {
+            if (!function_exists($callable)) {
+                return sprintf('Function "%s" does not exist.', $callable);
+            }
+
+            if (false !== strpos($callable, '::')) {
+                $callable = explode('::', $callable);
+            }
+        }
+
+        if (!is_array($callable)) {
+            return sprintf('Invalid type for controller given, expected string or array, got "%s".', gettype($callable));
+        }
+
+        if (2 !== count($callable)) {
+            return sprintf('Invalid format for controller, expected array(controller, method) or controller::method.');
+        }
+
+        list($controller, $method) = $callable;
+
+        if (is_string($controller) && !class_exists($controller)) {
+            return sprintf('Class "%s" does not exist.', $controller);
+        }
+
+        $reflection = new \ReflectionClass($controller);
+
+        if (method_exists($controller, $method)) {
+            return sprintf('Method "%s" on class "%s" should be public and non-abstract.', $method, $reflection->getName());
```

Comment:
Why using a ReflectionClass just to get the class name ? Use ``get_class($controller)`` when it is an object, and ``$controller`` when it is a string